pubnub/lib/src/core/logging/logging.dart

```diff
@@ -130,3 +175,189 @@ abstract class Level {
         .value;
   }
 }
+
+Map<String, dynamic> extractObjectProperties(Object obj,
+    {List<String> skipProperties = const ['keyset']}) {
+  final result = <String, dynamic>{};
+
+  try {
+    final mirror = reflect(obj);
+    final declarations = mirror.type.declarations;
+
+    for (var declaration in declarations.values) {
+      if (declaration is VariableMirror && !declaration.isStatic) {
+        final name = MirrorSystem.getName(declaration.simpleName);
+
+        if (skipProperties.contains(name)) {
+          continue;
+        }
+
+        try {
+          final value = mirror.getField(declaration.simpleName).reflectee;
+          result[name] = value;
+        } catch (e) {
+          result[name] = '<inaccessible>';
+        }
+      }
+    }
+  } catch (e) {
+    // Fallback: try to convert to string representation
+    result['toString'] = obj.toString();
+  }
+
+  return result;
+}
+
+/// Enum for different types of log event details
+enum LogEventDetailsType {
+  pubNubInstanceInfo,
+  apiParametersInfo,
+  networkRequestInfo,
+  networkResponseInfo,
+}
+
+/// Enhanced LogEvent that can handle both Map and Object details
+class LogEvent {
+  final String? message;
+  final Object? details;
+  final LogEventDetailsType? detailsType;
+
+  LogEvent({this.message, this.details, this.detailsType});
+
+  @override
+  String toString() {
+    var messageString = message ?? '';
+    try {
+      if (detailsType == LogEventDetailsType.pubNubInstanceInfo) {
+        var pubnub = details as PubNub;
+        var keysets = pubnub.keysets;
+
+        messageString += '\n\tPubNub Instance Information:';
+        messageString += '\n\tInstance ID: ${Core.instanceId}';
+        messageString += '\n\tVersion: ${Core.version}';
+
+        // Log module information
+        messageString += '\n\tModules:';
+        messageString +=
```

Comment:
I see that each module could override `toString()` method, which should make creating a final String representation of a `PubNub` object a bit easier:

```
messageString += pubnub.networking.toString()
messageString += pubnub.parser.toString()
messageString += pubnub.crypto.toString()
```